blob/service.go

```diff
@@ -251,6 +251,18 @@ func (s *Service) getByCommitment(
 
 	getCtx, getSharesSpan := tracer.Start(ctx, "get-shares-by-namespace")
 
+	// ensure that requested namespace can be found inside the DAH before retrieving it.
+	offset := -1
+	for i, row := range header.DAH.RowRoots {
+		if !namespace.IsOutsideRange(row, row) {
+			offset = i
+			break
+		}
+	}
+	if offset == -1 {
+		return nil, nil, ErrBlobNotFound
+	}
+
```

Comment:
Why? GetSharesByNamespace does this already iirc

Reply:
I think the idea here is to find offset, while it also allows to short circuit early. As there is demand to return index of commitment, I assume it is a matter of time we would want to add row index to `NamespacedRow` returned by `GetSharesByNamespace`. 

Lets add Idx field to `NamespacedRow`, so it can accessed it here and in `GetSharesByNamespace` rpc as well.

Reply:
IIRC, the proof from nmt contains the start idx in the row so we can add index method there already.

Reply:
I mean that we can add `rowIdx` to `NamespacedRow`. Using rowIdx and proof.Start index it will be trivial to calculate commitment start index

Reply:
The proof contains the start idx of the whole namespace row, which can contain multiple blobs + there might be padding shares.